File: languages/java/src/main/java/com/bloock/sdk/entity/EcdsaKeyPair.java
package com.bloock.sdk.entity;

import com.bloock.sdk.bridge.proto.Authenticity.GenerateEcdsaKeysResponse;

public class EcdsaKeyPair extends KeyPair {
  EcdsaKeyPair(String publicKey, String privateKey) {
    super(publicKey, privateKey);
  }

  public static EcdsaKeyPair fromProto(GenerateEcdsaKeysResponse keys) {
    return new EcdsaKeyPair(keys.getPublicKey(), keys.getPrivateKey());
  }
}

File: languages/php/Bloock/Entity/EcdsaKeyPair.php
<?php

namespace Bloock\Entity;

use Bloock\GenerateEcdsaKeysResponse;

class EcdsaKeyPair extends KeyPair
{
    public function __construct(string $publicKey, string $privateKey)
    {
        parent::__construct($publicKey, $privateKey);
    }

    public static function fromProto(GenerateEcdsaKeysResponse $res): EcdsaKeyPair {
        return new EcdsaKeyPair($res->getPublicKey(), $res->getPrivateKey());
    }
}